### PR TITLE
Quick summary

### LINKED ISSUE #135: Query data for graph rather than collect all
A much more scalable solution to the graph would be to query the data necessary to fulfill it rather than collect all the logs then do client side calculations. Doing all the hard work server side will make the client much faster. Also easier to modify. This will require modifications to the Logless server.

### PR DESCRIPTION
**Pull request by chrsdietz**
Fixes #135 

Changes in this pull request include:
- SourcePage now queries all stats data on load.

### CODE REVIEW

File: src/pages/SourcePage.test.tsx [hunk 1]
```diff
@@ -1,45 +1,283 @@
...
+
+            wrapper = shallow(<SourcePage source={undefined} />);
+            wrapper.setProps({
+                source: source
+            });
+            console.info("UPDATE");
+            wrapper.update();
+        });
+
+        // beforeEach(function () {
```
Comment (michaelmyers) [inline src/pages/SourcePage.test.tsx]: Remove these if no longer needed.

File: src/components/Graphs/Bar/CountChart.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import * as chai from "chai";
+import { shallow, ShallowWrapper} from "enzyme";
+import * as React from "react";
+
```
Comment (michaelmyers) [inline src/components/Graphs/Bar/CountChart.test.tsx]: Combine these import groups by removing this line.

File: src/components/Graphs/Line/TimeChart.test.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import * as chai from "chai";
+import { shallow, ShallowWrapper } from "enzyme";
+import * as React from "react";
+
```
Comment (michaelmyers) [inline src/components/Graphs/Line/TimeChart.test.tsx]: Combine these two 3rd party import groups by removing this line.

File: src/components/Graphs/Line/TimeChart.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import * as moment from "moment";
+import * as React from "react";
+
```
Comment (michaelmyers) [inline src/components/Graphs/Line/TimeChart.tsx]: Combine these import groups by removing this line.

File: src/models/query.test.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import * as chai from "chai";
+
+import * as Q from "./query";
```
Comment (michaelmyers) [inline src/models/query.test.ts]: [No wildcard imports](https://github.com/airbnb/javascript#modules--no-wildcard), even in tests.

File: src/pages/SourcePage.tsx [hunk 6]
```diff
@@ -5,23 +5,34 @@ import { Link } from "react-router";
...
+import TimeChart, { TimeData } from "../components/Graphs/Line/TimeChart";
 import { Cell, Grid } from "../components/Grid";
-import  SourceSummaryView from "../components/SourceSummaryView";
-import ConversationList from "../models/conversation-list";
-import ConversationListSummary from "../models/conversation-list-summary";
 import Source from "../models/source";
 import { State } from "../reducers";
-import { LogMap } from "../reducers/log";
-import DateUtils from "../utils/date";
+
```
Comment (michaelmyers) [inline src/pages/SourcePage.tsx]: Remove whitespace to combine with the above import group.  Or is it separate for a reason?

**Comment by chrsdietz** [comment]
Separating the import groups wasn't intentional. I think it was a newline added by copy and pasting.

**Pull request merged**